src/firefly/java/edu/caltech/ipac/firefly/server/visualize/FitsCacher.java

```diff
@@ -95,12 +94,18 @@ private static void addToCache(FileInfo fitsFileInfo, FitsDataEval fitsDataEval)
         if (memCache != null) memCache.put(fitsFileInfo, fitsDataEval);
     }
 
+    public static void refreshCache(File fitsFile) {
+        FileInfo fitsFileInfo= getFileInfoFromCache(fitsFile);
+        FitsDataEval fitsDataEval= getFromCache(fitsFileInfo);
+        if (fitsDataEval==null) return;
+        addToCache(fitsFileInfo,fitsDataEval);
+    }
+
 
     private static void prepareCacheSpace(FileInfo fitsFileInfo) {
-        if (memCache != null) {
-            memCache.put(fitsFileInfo, new ObjectSizeEngineWrapper.BluffSize(fitsFileInfo.getFile().length()));
-            memCache.put(fitsFileInfo, null);
-        }
+        if (memCache == null) return;
+        memCache.put(fitsFileInfo, new ObjectSizeEngineWrapper.BluffSize(fitsFileInfo.getFile().length()));
```

Comment:
A good place to use anonymous class.  BluffSize is no longer needed.
`memCache.put(fitsFileInfo, (HasSizeOf) () -> fitsFileInfo.getFile().length());`

Reply:
I did not realize you could use a cast for force a certain type of interface.